### ISSUE TITLE
Form Request does not contain files in $this->getInputSource()

### ISSUE THREAD

**Issue by terion-name** [laravel/internals#476]
Laravel 5.4.15

I send `FormData` with js from my app to Laravel backend via `post`. Request:
```
Accept:application/json
Accept-Encoding:gzip, deflate
Accept-Language:uk
Cache-Control:no-cache
Connection:keep-alive
Content-Length:133752
Content-Type:multipart/form-data; boundary=----WebKitFormBoundaryALUi6UOHwUbyuoAy
Cookie:...
Origin:http://homestead.app
Pragma:no-cache
Referer:http://homestead.app/app/cabinet/profile
User-Agent:Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/56.0.2924.87 Safari/537.36
X-CSRF-TOKEN:mTrII2awr23gUIFPCl2Odg1jpxLXoEguM47ShZKR
X-Requested-With:XMLHttpRequest
```

Request is precessed with `Form Request`, but I don't get the file included.

If I run in FormRequest the following:
```php
protected function prepareForValidation() {
        dd($this->request->all()); // or dd($this->getInputSource()->all())
}
```
I get data without the file.

HOWEVER
This:
```php
protected function prepareForValidation() {
        dd(request()->all());
}
```
Prints data with `UploadedFile` properly

**Comment by salomoni** [comment]
The `FormRequest` is the request so 

```php
protected function prepareForValidation() {
    dd($this->all());
}
```

**Comment by terion-name** [comment]
@salomoni oh, ok, but.
There is a hook `prepareForValidation`. It is intended to transform input. I use it for filtering input — parse validation rules and leave only the fields that are acceptable.
This is possible via `$this->getInputSource()`
```php
// fully reconstruct input
foreach ($this->getInputSource()->keys() as $key) {
            $this->getInputSource()->remove($key);
        }
        foreach ($inputFiltered as $k => $v) {
            $this->getInputSource()->set($k, $v);
        }
```

And there is to things:
1. `$this->getInputSource()` does not contain files
2. after this operation the request object looses files, despite that `$this->files` was not touched

**Comment by salomoni** [comment]
Either you are removing the file explicitly or it has not been sent along with the request at all.

Do you get the `UploadedFile` when you `dd($this->all())` in the `prepareForValidation`?

**Comment by terion-name** [comment]
Yes. `dd($this->all())` prints file.

Here is the code:

```php
<?php

namespace App\Http\Requests;

use Illuminate\Foundation\Http\FormRequest;

abstract class FilteredFormRequest extends FormRequest
{
    /**
     * Prepare the data for validation.
     * Leave only data specified in validation rules
     *
     * @return void
     */
    protected function prepareForValidation()
    {
        // get the input
        $input = array_filter($this->getInputSource()->all(), function ($v) {
            return !is_null($v);
        });
        // flatten it to single-level dot-array to match validation rules syntax
        $inputFlattened = array_dot($input);
        // rules keys
        $validated = array_keys($this->rules());
        // get only validated input respecting the asterisk
        $inputFiltered = [];
        foreach ($inputFlattened as $key => $value) {
            foreach ($validated as $vk) {
                $regex = '/^' . str_replace('\*', '[^\.]*', preg_quote($vk)) . '$/';
                if (preg_match($regex, $key)) {
                    // set in expanded form
                    array_set($inputFiltered, $key, $value);
                }
            }
        }
        // fully reconstruct input
        foreach ($this->getInputSource()->keys() as $key) {
            $this->getInputSource()->remove($key);
        }
        foreach ($inputFiltered as $k => $v) {
            $this->getInputSource()->set($k, $v);
        }
    }
}
```

And when payload contains a file (e.g. `logo`), after this in controller method `$request->get('logo')` return null, as well as `$request->files` 

Currently, as a workaround, I get base input from ´$this->all()´, but this seems to be strange.

Also, if I remember correctly, there was no such issue earlier. As I think, this behavior appeared after framework update from version, where `ConvertEmptyStringsToNull` middleware was not applied to `FormRequests` (prior 5.4.9) to a version where this was fixed (after 5.4.9, If I remember correctly, I've updated from 5.4.8 to 5.4.13 then)

**Comment by salomoni** [comment]
`$request->file('logo')` to get params from the `FileBag`. `$request->get('logo')` searches from the `ParameterBag`.

v5.4.14 and v.5.4.15 works fine for me using the exact same `prepareForValidation` as you. `ConvertEmptyStringsToNull` middleware didn't seem to affect this.

Do `dd` line by line until you break it.
